media-video/mjpg-streamer/mjpg-streamer-0_pre20120621-r2.ebuild

```diff
@@ -13,37 +13,43 @@ LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
 
-INPUT_PLUGINS="input_testpicture input_control input_file input_uvc"
-OUTPUT_PLUGINS="output_file output_udp output_http output_autofocus output_rtsp"
+INPUT_PLUGINS="input-testpicture input-control input-file input-uvc"
+OUTPUT_PLUGINS="output-file output-udp output-http output-autofocus output-rtsp"
 IUSE_PLUGINS="${INPUT_PLUGINS} ${OUTPUT_PLUGINS}"
-IUSE="input_testpicture input_control +input_file input_uvc output_file
-	output_udp +output_http output_autofocus output_rtsp
+IUSE="input-testpicture input-control +input-file input-uvc output-file
+	output-udp +output-http output-autofocus output-rtsp
 	www v4l"
 REQUIRED_USE="|| ( ${INPUT_PLUGINS} )
 	|| ( ${OUTPUT_PLUGINS} )
-	v4l? ( input_uvc )"
+	v4l? ( input-uvc )"
 
 RDEPEND="virtual/jpeg
-	v4l? ( input_uvc? ( media-libs/libv4l ) )"
+	v4l? ( input-uvc? ( media-libs/libv4l ) )"
 DEPEND="${RDEPEND}
-	input_testpicture? ( media-gfx/imagemagick )"
+	input-testpicture? ( media-gfx/imagemagick )"
+
+PATCHES=(
+	"${FILESDIR}/make-var-instead-of-cmd.patch"
+	"${FILESDIR}/to-work-with-kernel-3.18.patch"
+)
 
 src_prepare() {
-	epatch "${FILESDIR}/${PV}-make-var-instead-of-cmd.patch"
-	epatch "${FILESDIR}/${PV}-to-work-with-kernel-3.18.patch"
+	default
 
 	local flag switch
 
 	for flag in ${IUSE_PLUGINS}; do
 		use ${flag} && switch='' || switch='#'
+		flag=${flag/input-/input_}
+		flag=${flag/output-/output_}
 		sed -i \
 			-e "s|^#*PLUGINS +\?= ${flag}.so|${switch}PLUGINS += ${flag}.so|" \
 			Makefile
 	done
 }
 
 src_compile() {
-	local v4l=$(use v4l && use input_uvc && echo 'USE_LIBV4L2=true')
+	local v4l=$(use v4l && use input-uvc && echo 'USE_LIBV4L2=true')
```

Comment:
what v4l does without input-uvc?

Reply:
there was one plugin input_gspcav1 using v4l, but since that is no longer built as plugins, we can drop this instead and use only input-uvc as USE requiring media-libs/libv4l